src/services/ApiService.ts

```diff
@@ -0,0 +1,72 @@
+import { groupBy } from 'lodash';
+import { ActiveFilterData } from '@/types/apiFilters';
+import HttpService from '@/services/HttpService';
+import { ApiMetadata } from '@/types/api';
+import { ApiDefinition } from '@/types/apiDefinition';
+import { ApiVersion } from '@/types/apiVersion';
+import { ApiDeployment } from '@/types/apiDeployment';
+import { ApiEnvironment } from '@/types/apiEnvironment';
+
+const ApiService = {
+  async getApis(search: string, filters: ActiveFilterData[] = []) {
+    const searchParams = new URLSearchParams();
+    if (search.length) {
+      searchParams.set('$search', search);
+    }
+
+    if (filters.length) {
+      const filtersByType = groupBy(filters, 'type');
+      const filtersString = Object.values(filtersByType)
+        .map((filters) => {
+          const filtersSet = filters.map((filter) => `${filter.type} eq '${filter.value}'`);
+          return `(${filtersSet.join(' or ')})`;
+        })
+        .join(' and ');
+
+      searchParams.set('$filter', filtersString);
+    }
+
+    const response = await HttpService.get<{ value: ApiMetadata[] }>(`/apis?${searchParams.toString()}`);
+    return response.value || [];
+  },
+
+  async getApi(id: string) {
+    return await HttpService.get<ApiMetadata>(`/apis/${id}`);
+  },
+
+  async getVersions(apiId: string) {
```

Comment:
Please add return types whenever possible. It makes easier for consumers (and this is a public repo) to understand the code.

Reply:
Modern IDEs usually resolve return types for you but sure, I can add them explicitely.
![image](https://github.com/user-attachments/assets/c5bb9c5b-c058-4755-898d-d0498e6fe574)